pcsx2/gui/AppCoreThread.cpp

```diff
@@ -427,23 +427,29 @@ static void _ApplySettings( const Pcsx2Config& src, Pcsx2Config& fixup )
 		// the BIos isn't loaded until after initial calls to ApplySettings.
 
 		gameName = L"Booting PS2 BIOS... ";
-	}
 
 	//Till the end of this function, entry CRC will be 00000000
-	if (!gameCRC.Length())
+	CDVD_SourceType cdvdsrc(g_Conf->CdvdSource);
```

Comment:
You could convert this line to a bool for checking `cdvdsrc != CDVD_SourceType::NoDisc`.